### PR TITLE
164 166 170 Add source data folder and improve pathing to package root

### LINKED ISSUE #122: Adjust OSM utils tests so that they are not OS specific.
## Description
When running the test suite on a windows machine, the test in `tests/osm/test_osm_utils.py` fails due to the test being OS specific (to MacOS).  
More specifically, the tests is calling the `filter_osm()` function with an invalid path, therefore expecting a `FileExistsError` to be raised.  

The matching regex inside of `pytest.raises()` uses forward slashes, along with the raw string for the path, therefore failing on windows as once converted to a `Pathlib.Path` object, back slashes are used.

## Screenshots  
_Test that fails_

Image

_Test result_

Image

## Suggestions  
Possible fixes to this test include:  
1. Using `os.path.join()` for the path passed to `filter_osm()`, along with the matching regex.
#### Example  
```python
with pytest.raises(
            FileExistsError, match=f"{os.path.join("not", "a", "pbf", ".nosiree")} not found on file."
        ):
            # file doesnt exist
            filter_osm(pbf_pth=os.path.join("not", "a", "pbf", ".nosiree"))
```

## Test Conditions

OS: Windows 10
Python Version: 3.9.13
Environment: conda

### LINKED ISSUE #164: get_saved_route_type_lookup() doesn't work when not called from project root
**Describe the bug**

When instantiating `GtfsInstance` from a file not at the directory root, say from within a file in a sub-directory or an entirely different directory, a `FileNotFound` error is raised when `get_saved_route_type_lookup()` tries to read `tests/data/gtfs/route_lookup.pkl`. This cause by `tests/data/gtfs/route_lookup.pkl` becoming an absolute path relative to where the function is called from.

**Environment**

 - OS: macOS
 - Python version: 3.9.13
 - Java version: 11
 - Python module versions (if different from `requirements.txt`):

**Steps to reproduce**

1. Create a .py file in another directory other than the root.
2. Import `GtfsInstance` and instantiate it with a GTFS file
3. A `FileNotFound` error will be raised, sharing a directory and file that doesn't exist (assumes `tests/data/...` exists at the same level as the .py file).

**Expected behavior**

`GtfsInstance` can be instantiated from anywhere

**Screenshots**

**Test data/code**

**Additional context**

This pickle file is essentially a package resource, so perhaps we should use [`importlib.resources`](https://docs.python.org/3.11/library/importlib.resources.html) to handle things. That way, it will read from the location where the package has been installed and expected the `tests/` directory to exist in the same dir.

### LINKED ISSUE #166: Create optional 'route_lkp_pth' argument for GtfsInstance
### Description
After reviewing the use of `get_scraped_route_lookup()` in `GtfsInstance`, it is clear that the user should be able to specify a path for the lookup.  
Current behaviour uses a hard coded path

### LINKED ISSUE #170: Create a data folder the the package to be included with package install (pip)
**Description of the Feature to be Added**

In order for data files to be used within the project, they must be compiled when the package is pip installed. Certain test fixtures and other non-code files were previously used by the package however would not be included when the package was installed.  

This requires a data folder to be made within the source.  

This issue became apparent as a result of issue #164

### PR DESCRIPTION
**Pull request by cbrowning04**
## Description

This PR includes multiple improvements to the package,  including fixing bugs that stopped a user without the entire repo from using the functionality of this package.  

Summarised Changes:
- Added a data directory in `src/transport_performance`.
- Moved data files (non-code) that the package depends on into the data directory and removed them from their previous locations.
- Added a `MANIFEST.in` file to ensure the correct data files are being included in the package.
- Added a `constants.py` file that allows for easier use of constants in a centralised area.
- The `PKG_PTH` constant has been adopted across the package in areas where data files are used, ensuring the code still functions after installing only the package.
- `GtfsInstance()` now has an additional parameter `route_lookup_pth`. This allows for the user to specify their own route type lookup rather than forcing them to use the default one (currently saved in the data folder).
- Fixed an OSM test that was failing on windows. This was a small change and only required some changes to the formatting of paths in raised errors.

Fixes #166 
Fixes #170 
Fixes #164 
Fixes #122 

## Motivation and Context

The motivation of this PR is to ensure that the `transport_performance` package can be successfully installed through pip without leaving the user with missing data (that is required to use some of the main functions of the package).  
There are also updates that give the user more flexibility, such as the new parameter in `GtfsInstance()`.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Test configuration details:
* OS: Windows 10
* Python version: 3.9.13
* Java version: N/A
* Python management system: Conda

## Advice for reviewer

To learn more about the use of the data folder and `MANIFEST.in`, read the first section of [this page](https://setuptools.pypa.io/en/latest/userguide/datafiles.html).

## Checklist:

- [x] My code follows the intended structure of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Additional comments

Issues raised as a result of this PR include:
#169

### CODE REVIEW

**Review by r-leyshon — CHANGES_REQUESTED** [review]
Hi @CBROWN-ONS , before I get into the line by line stuff, I've realised that it would be best to discuss one file structure change first:
I feel differently about newport-20230613_gtfs.zip than I do route_lookup.pkl. The GTFS is much larger thank the pickle, nearly a mb. 
The pickle is used in `gtfs.validation.GtfsInstance._plot_summary()` for a real-world use, here building a table used for presenting readable route type statistics in the GTFS validation HTML outputs. This file is therefore crucial to the functionality of the api.
The GTFS file; in contrast; was included with the intention as a test fixture only. The fact that default paths point at it would be for developer convenience reasons only. Packaging this fixture would be of almost no use to users of our API, unless they had a curious interest in the specific bounding box around Newport.
Rather than including it in the src folder, I would opt to remove default file paths pointing at the fixture and refactor the test suites. Hope that makes sense, happy to discuss with the broader team if necessary. In the same way that "tests/data/newport-2023-06-13.osm.pbf” is treated.
Files impacted by this suggested change:
* tests/gtfs/test_gtfs_utils.py
* tests/gtfs/test_routes.py
* tests/gtfs/test_validation.py
* tests/osm/test_osm_utils.py
* tests/utils/test_defence.py
* src/transport_performance/gtfs/validation.py
* src/transport_performance/gtfs/routes.py
* src/transport_performance/gtfs/gtfs_utils.py
* src/transport_performance/data/gtfs/newport-20230613_gtfs.zip
* .gitignore

**Comment by cbrowning04** [comment]
> Hi @CBROWN-ONS , before I get into the line by line stuff, I've realised that it would be best to discuss one file structure change first:
> 
> I feel differently about newport-20230613_gtfs.zip than I do route_lookup.pkl. The GTFS is much larger thank the pickle, nearly a mb. 
> 
> The pickle is used in `gtfs.validation.GtfsInstance._plot_summary()` for a real-world use, here building a table used for presenting readable route type statistics in the GTFS validation HTML outputs. This file is therefore crucial to the functionality of the api.
> 
> The GTFS file; in contrast; was included with the intention as a test fixture only. The fact that default paths point at it would be for developer convenience reasons only. Packaging this fixture would be of almost no use to users of our API, unless they had a curious interest in the specific bounding box around Newport.
> 
> Rather than including it in the src folder, I would opt to remove default file paths pointing at the fixture and refactor the test suites. Hope that makes sense, happy to discuss with the broader team if necessary. In the same way that "tests/data/newport-2023-06-13.osm.pbf” is treated.
> 
> Files impacted by this suggested change:
> 
> * tests/gtfs/test_gtfs_utils.py
> 
> * tests/gtfs/test_routes.py
> 
> * tests/gtfs/test_validation.py
> 
> * tests/osm/test_osm_utils.py
> 
> * tests/utils/test_defence.py
> 
> * src/transport_performance/gtfs/validation.py
> 
> * src/transport_performance/gtfs/routes.py
> 
> * src/transport_performance/gtfs/gtfs_utils.py
> 
> * src/transport_performance/data/gtfs/newport-20230613_gtfs.zip
> 
> * .gitignore

Hi Rich. Me and Ethan had previously discussed this but had waited on implementing it before discussing it with the rest of the team. I 100% agree that we should remove default file paths to the Newport test fixture. I do think it might be more useful to raise as a feature request for now however as it would be a breaking change (I believe).

**Comment by r-leyshon** [comment]
One other point to note, failing test on --runsetup flag:
```
tests/test_setup.py::TestSetup::test_r5py_setup FAILED                                                                [  0%]
```
Line 43 is searching in the test fixtures dir `  search_pth = os.path.join("tests", "data")`

File: pyproject.toml [hunk 1]
```diff
@@ -19,6 +19,12 @@ classifiers = [
 requires-python = ">=3.9, <3.10"
 version = "0.0.1"
 
+[tool.setuptools]
+# This is true by default so this can be removed.
+# Currently retaining this line before discussing with others whether or not
+# we want to keep it for more clarity.
+include-package-data = true
```
Comment (r-leyshon) [inline pyproject.toml:26]: agreed that [setuptools docs](https://setuptools.pypa.io/en/stable/userguide/datafiles.html) suggest this is on as default.

**Comment by cbrowning04** [comment]
> One other point to note, failing test on --runsetup flag:
> 
> ```
> 
> tests/test_setup.py::TestSetup::test_r5py_setup FAILED                                                                [  0%]
> 
> ```
> 
> Line 43 is searching in the test fixtures dir `  search_pth = os.path.join("tests", "data")`

This should be resolved once the test fixtures are moved back to the tests folder and default file paths for Newport fixtures are removed from the Gtfs module

**Comment by cbrowning04** [comment]
HI @r-leyshon. I've just had a look over the changes and tested them locally (pytest + external test of `GtfsInstance()`). Everything is working as expected, and from reading through the code it seems like we have met what we set out to achieve with making the `gtfs_pth` parameter required.  
Happy for this to be merged if you are!

**Review by r-leyshon — APPROVED** [review]

**Pull request merged**